apps/astarte_realm_management_api/test/astarte_realm_management_api/devices/devices_test.exs

```diff
@@ -18,22 +18,31 @@
 
 defmodule Astarte.RealmManagement.API.DevicesTest do
   use Astarte.RealmManagement.API.DataCase
+  use ExUnitProperties
 
   alias Astarte.RealmManagement.API.Devices
   alias Astarte.RealmManagement.API.Helpers.RPCMock.DB
+  alias Astarte.Core.Generators.Device, as: DeviceGenerator
+  alias Astarte.Core.Device
 
   @realm "testrealm"
 
-  test "delete device succeeds when the device exists" do
-    device_id = :crypto.strong_rand_bytes(16) |> Base.url_encode64(padding: false)
-    DB.create_device(@realm, device_id)
+  describe "property based device tests" do
+    @describetag :devices
+    property "delete device succeeds when the device exists" do
+      check all(device <- DeviceGenerator.device(interfaces: [])) do
+        encoded_device_id = Device.encode_device_id(device.device_id)
```

Comment:
Here the device generator has the generator for an `encoded_device_id` directly, no need to generate an entire device :+1: 

```suggestion
      check all(encoded_device_id <- DeviceGenerator.encoded_id()) do
```